python_modules/libraries/dagster-dg/dagster_dg/config.py

```diff
@@ -454,25 +449,90 @@ def _raise_file_config_validation_error(message: str, file_path: Path) -> Never:
     raise DgError(f"Error in configuration file {file_path}: {message}")
 
 
-def _get_types_str(type_: Union[type, tuple[type, ...]]) -> str:
-    types = type_ if isinstance(type_, tuple) else (type_,)
-    return " or ".join(_get_type_str(t) for t in types)
-
-
-# now way to type Literal
-def _get_literal_values(type_: Any) -> tuple[object, ...]:
-    val = type_
-    while get_origin(val) is not Literal:
-        val = get_origin(val)
-    return get_args(val)
+# expected_type Any to handle typing constructs (`Literal` etc)
+def get_type_str(t: Any) -> str:
```

Comment:
what is all this and why is it necessary?

Reply:
Seems like overkill. Isn't this what pydantic is for?

Reply:
Let's remove this for now and find a more holistic solution for `toml` error reporting later. We should add it deliberately in a well-tested way.

Reply:
We don't depend on `pydantic` in `dagster-dg`.

Reply:
Would prefer to rely on Pydantic than to roll our own thing here.

Reply:
Discussed offline. This is partially a function of the fact that we don't have a shared package between `dagster` and `dagster-dg`. We will at a minimum consolidate this code with similar code that exists in systems like `check` or adopt a more out-of-the-box solution later.